### DISCUSSION TITLE
[V4.1] Is transaction supported in all of the relationship methods ?

### DISCUSSION THREAD

**Discussion by mjsy** [adonisjs/core#1635] [Help] [answered]
Can anyone confirm me that transactions work with `save`, `create`, `associate`, `dissociate` and other methods on a relationship ? I can't find this in the docs except for the `attach` and `detach` method which take the transaction as third param.

Example : 
`await user.posts().create({ rating: body.rating }, trx)`
`await post.media().save(media, trx)`
`await profile.avatar().associate(media, trx)`

**Comment by thetutlage** [comment, marked as answer]
It does. Here's the reference to the code  https://github.com/adonisjs/lucid/blob/210e73904317516d40a71176c3f4dfaee619b427/src/Lucid/Relations/BelongsTo.js#L180

**Reply by mjsy** [reply]
Perfect, thank you !
